SwiftyJSONAPI/JSONAPIResource.swift

```diff
@@ -103,10 +104,14 @@ public class JSONAPIResource: JSONPrinter {
                 resource.attributes    = includedResource.attributes
                 resource.relationships = includedResource.relationships
                 
-                if !resource.relationships.isEmpty {
+                if !resource.relationships.isEmpty, !cachedResourceIds.contains(resource.id) {
+                    
+                    cachedResourceIds.insert(resource.id)
```

Comment:
Shouldn't cached resources be keyed by both `type` and `id`, since the included section can contain many different kinds of resources.  I could have a `type=user` with `id=1` as well as an `type=order` with `id=1` (albeit unlikely).  In general, a resource in json api is always uniquely by its `type` and `id`

Reply:
Yup, @gtrog has a point. This should be by type and id like the included resources.